uk.co.compucorp.civicrm.hrleaveandabsences/views/shared/directives/leave-request-popup.html

```diff
@@ -203,32 +203,29 @@ <h4 class="modal-title">Leave Request</h4>
                 </div>
               </div>
             </crm-loading>
-            <div class="row" ng-show="$ctrl.sickness">
+            <div class="row" ng-show="$ctrl.isleaveType('sick')">
               <div class="col-xs-12 form-horizontal">
               <hr>
                 <div class="form-group">
                   <label class="col-xs-4 control-label">Reason<i class="fa fa-asterisk"></i></label>
                   <div class="col-xs-8">
                     <div class="chr_custom-select chr_custom-select--full">
-                      <select name="reason">
-                        <option value="- select -">- select -</option>
+                      <select
+                        name="reason"
+                        ng-options="reason.name as reason.label for reason in $ctrl.leaveRequestReasons"
+                        ng-model="$ctrl.leaveRequest.reason">
+                        <option value="">- select -</option>
                       </select>
                     </div>
                   </div>
                 </div>
                 <div class="form-group">
                   <div class="col-xs-12">
-                    <div>
-                      <input required type="checkbox" value="certification">
-                      <span>Self certification form required</span>
-                    </div>
-                    <div>
-                      <input required type="checkbox" value="interview">
-                      <span>Back to work interview required</span>
-                    </div>
-                    <div>
-                      <input required type="checkbox" value="work">
-                      <span>Return to work required</span>
+                    <div ng-repeat="document in $ctrl.leaveRequestDocuments">
+                      <input ng-disabled="$ctrl.isRole('owner')" required type="checkbox" value="document.value"
+                        ng-click="$ctrl.updateRequiredDocuments(document.value)"
```

Comment:
1. there is no default array of `required_documents`
2. there would be 2 arrays or arrays of objects and its conversion to `required_documents` array if I go by your suggestion